### ISSUE TITLE
Make the `color = dims(1)` correspond to the `mapping` for wide data

### ISSUE THREAD

**Issue by pdeffebach** [MakieOrg/AlgebraOfGraphics.jl#474]
## Problem description

I am making a line plot with wide data. The command is

```
p = data(df_collapsed) * mapping(:mean_age, [:N, :U, :B, :W], color = dims(1) => renamer(["N", "U", "B", "W"])) * visual(Lines)
```

See how I have to repeat `["N", "U", "B", "W"]` twice? I should just have to write `dims(1)` and have it automatically take the name of the lines from the `mapping`.

**Comment by pdeffebach** [comment]
Here is an MWE

```
julia> using AlgebraOfGraphics, GLMakie, DataFrames;

julia> df = DataFrame(x = 1:10, y1 = rand(10) .+ 1, y2 = rand(10) .+ 2);

julia> p = data(df) * mapping(:x, [:y1, :y2], color = dims(1)) * linear();

julia> draw(p)
```

Which gives

![image](https://github.com/MakieOrg/AlgebraOfGraphics.jl/assets/23196228/8c27cc4c-49c3-4850-9349-cbc05b82b0d8)

I would like the two lines on the graph to be labelled "y1" and "y2" rather than `CartesianIndex` as is the current behavior.

**Comment by pdeffebach** [comment]
Bumping this, as it would be a nice feature to add.

**Comment by jkrumbiegel** [comment]
I'm not 100% sure if it's feasible but I could try. The basic problem is that the dimensions are not necessary from wide columns like `[:y1, :y2]`, it could also be passed directly via `pregrouped` for example. Or you could have multiple wide data inputs, in which case names would be ambiguous. But that of course doesn't mean that it would be impossible to support this basic case, which is probably the most common.

**Comment by pdeffebach** [comment]
I wasn't aware of `pregrouped`, but that seems like it has a pretty different code path?

Yeah, I just think this is nice to have. A lot of economists (me included) prefer wide to long data, and Stata has nice default behavior on this front.

**Comment by jkrumbiegel** [comment]
The code path for pregrouped is a bit different yes, but the dimensions that `dims` picks up on are determined across the full layer. For example if you do `mapping([:a, :b, :c], [:d, :e, :f], color = dims(1))` then the columns for x and y are different but they both span up the same dimensional shape of `(3,)`.

What I'm saying is, maybe one could add a convenience code path where we determine all inputs that have dimensions > 1 that contribute to a given `dims`, and if there's just one, then we use the labels from those or something. I also never liked the `CartesianIndex` thing, it's just a side effect of the implementation, I don't think it was ever considered a nice UX.

**Comment by pdeffebach** [comment]
I'm not sure I understand. This should still give `:x` and `:y` for the colors, right?

```
julia> df = DataFrame(a = 1:10, b = 11:20, x = rand(10), y = rand(10))
10×4 DataFrame
 Row │ a      b      x          y         
     │ Int64  Int64  Float64    Float64   
─────┼────────────────────────────────────
   1 │     1     11  0.911835   0.948568
   2 │     2     12  0.670913   0.394753
   3 │     3     13  0.950907   0.760813
   4 │     4     14  0.752873   0.0232649
   5 │     5     15  0.0167365  0.205273
   6 │     6     16  0.172779   0.650877
   7 │     7     17  0.925592   0.228989
   8 │     8     18  0.379768   0.723976
   9 │     9     19  0.19861    0.839134
  10 │    10     20  0.0751635  0.628947

julia> data(df) * mapping([:a, :b], [:x, :y], color = dims(1)) * linear() |> draw
```

![Image](https://github.com/user-attachments/assets/bb6d7c3c-270e-4070-ad20-191592359cfe)

**Comment by jkrumbiegel** [comment]
No, it's `:a` and `:x` together and `:b` and `:y` together. Both wide inputs are vectors of length 2 so you get layer shape (2,). If you make one of them a row vector you should get for plots and (2,2) shape, then only one of them would correspond to `dims(1)` again

**Comment by pdeffebach** [comment]
Ah makes sense. 

Well I overall I think is a valuable feature and I support some special-casing to making this work.
